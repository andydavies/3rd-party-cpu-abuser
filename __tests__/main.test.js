const cpuAbuser = require('../index')
const mock = 'samples/VA-article.json'
const mockResult = require('./mock-output-VA-article.json')

jasmine.DEFAULT_TIMEOUT_INTERVAL = 15000

describe('Common behaviour', () => {
  it('should throw an error if file does not exist', () => {
    expect(() => cpuAbuser.data({
      file: 'non-existent'
    })).toThrow()
  })
})


let resultDefault

describe('Node API with JSON results : default options', () => {
  let result = cpuAbuser.data({
    file: mock
  })
  resultDefault = result // for next test to compare

  it('should have a JSON structure', () => {
    //console.log(result)
    expect( result.totalCPUTime ).toBeDefined()
    expect( result.cpuTimePerDomain ).toBeDefined()
  })

  it('timings should be Number', () => {
    //console.log(result)
    expect(result.totalCPUTime).toEqual(expect.any(Number))
    expect(result.cpuTimePerDomain['pagead2.googlesyndication.com']).toEqual(expect.any(Number))
  })

  it('should calculate totalCPUTime', () => {
    // console.log(result)
    expect(result.totalCPUTime).toBe(mockResult.totalCPUTime)
  })

  it('should calculate CPU time on a domain basis', () => {
    // console.log(result)
    expect(result.cpuTimePerDomain['pagead2.googlesyndication.com']).toBeDefined()
    expect(result.cpuTimePerDomain['pagead2.googlesyndication.com']).toBe(mockResult.cpuTimePerDomain['pagead2.googlesyndication.com'])
  })

  it('VA-article.json analysis should exactly match the saved results', () => {
    //console.log(result)
    expect(result).toEqual(mockResult)
    expect(result).toMatchSnapshot()
  })

})

describe('Node API with JSON results : minTime=10', () => {
  let result = cpuAbuser.data({
    file: mock,
    minTime: 10
  })

  it('Changing minTime should return more results', () => {
    expect(result.totalOffendersDomains).toBeGreaterThan(resultDefault.totalOffendersDomains)
  })

})

describe('Node API with JSON results : minTime=0', () => {
  let result = cpuAbuser.data({
    file: mock,
    minTime: 0
  })

  it('Changing minTime should return more results', () => {
    expect(result.totalOffendersDomains).toBeGreaterThan(resultDefault.totalOffendersDomains)
  })

})
