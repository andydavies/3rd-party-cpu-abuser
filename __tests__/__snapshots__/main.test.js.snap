exports[`Node API with JSON results VA-article.json analysis should exactly match the saved results 1`] = `
Object {
  "cpuTimePerDomain": Object {
    "acdn.adnxs.com": 729.0930000245571,
    "ajax.googleapis.com": 306.852000027895,
    "ak-ns.sascdn.com": 167.54000002145767,
    "cas.nl.eu.criteo.com": 495.8839999437332,
    "cdn.adnxs.com": 572.7570000141859,
    "cdn.taboola.com": 3756.774000287056,
    "cdn.teads.tv": 2898.4009999632835,
    "cstatic.weborama.fr": 282.5810000449419,
    "dmp.theadex.com": 185.69399999082088,
    "foglio.basilic.io": 184.6860000193119,
    "ib.adnxs.com": 178.52600000798702,
    "js.moatads.com": 164.74399998784065,
    "pagead2.googlesyndication.com": 1744.121000096202,
    "s.skimresources.com": 542.8169998824596,
    "s3.amazonaws.com": 166.26500004529953,
    "static.lexpress.fr": 394.9170000106096,
    "tpc.googlesyndication.com": 665.8419999629259,
    "votreargent.lexpress.fr.running.cloud.ger.systems": 3712.0569999367,
    "www.google-analytics.com": 450.12000001966953,
    "www6.smartadserver.com": 379.9199999421835,
  },
  "totalCPUTime": 18783.098000094295,
  "totalDomains": 51,
  "totalOffendersDomains": 20,
}
`;
